build/source/engine/allocspace.f90

```diff
@@ -67,11 +67,14 @@ subroutine allocate_gru_struc(nGRU,nHRU,err,message)
  USE summaFileManager,only:LOCAL_ATTRIBUTES        ! file containing information on local attributes
  USE globalData,only: index_map                    ! relating different indexing system
  USE globalData,only: gru_struc                    ! gru-hru mapping structures
+ use multiconst,only: integerMissing               ! value for missing integer
  
  implicit none
  ! define output
- integer(i4b),intent(out)             :: nGRU               ! number of grouped response units
- integer(i4b),intent(out)             :: nHRU               ! number of hydrologic response units
+ integer(i4b),intent(inout)           :: nGRU               ! number of grouped response units
+ integer(i4b),intent(inout)           :: nHRU               ! number of hydrologic response units
+ integer(i4b),intent(inout)           :: startGRU           ! index of the starting GRU for parallelization run
+ integer(i4b),intent(out)             :: maxGRU             ! maximum number of GRUs in the input file
```

Comment:
`maxGRU` : change to `intent(in)` and read `maxGRU` from file before entering this function